packages/react-packages/core/src/hooks/__tests__/useDebounceValue.test.tsx

```diff
@@ -0,0 +1,82 @@
+import { renderHook, act } from '@testing-library/react';
+
+import useDebounceValue from '../useDebounceValue';
+
+jest.useFakeTimers();
+
+describe('useDebounceValue', () => {
+  beforeEach(() => {
+    jest.clearAllTimers();
+  });
+
+  afterEach(() => {
+    jest.runOnlyPendingTimers();
+    jest.useRealTimers();
+    jest.useFakeTimers();
+  });
+
+  it('should return null initially', () => {
+    const { result } = renderHook(() => useDebounceValue('test', 500));
+    expect(result.current).toBe(null);
+  });
+
+  it('should debounce value changes', () => {
+    const { result, rerender } = renderHook(
+      ({ value }) => useDebounceValue(value, 500),
+      { initialProps: { value: 'initial' } }
+    );
+
+    expect(result.current).toBe(null);
+
+    act(() => {
+      jest.advanceTimersByTime(500);
+    });
+
+    expect(result.current).toBe('initial');
+
+    rerender({ value: 'updated' });
+    expect(result.current).toBe('initial'); // Should still be old value
+
+    act(() => {
+      jest.advanceTimersByTime(500);
+    });
+
+    expect(result.current).toBe('updated');
+  });
+
+  it('should handle null values correctly', () => {
+    const { result, rerender } = renderHook(
+      ({ value }) => useDebounceValue(value, 500),
+      { initialProps: { value: 'test' } }
+    );
+
+    act(() => {
+      jest.advanceTimersByTime(500);
+    });
+
+    expect(result.current).toBe('test');
+
+    rerender({ value: null });
+    expect(result.current).toBe('test'); // Should not change when value is null
```

Comment:
In this case, after rerender, should we do

```
act(() => {
      jest.advanceTimersByTime(500);
    });
```

Otherwise, the `result.current` is still the old value (as in https://github.com/daimlertruck/DT-DDS/pull/39/commits/25021c72e076216faf3bf804d9884b596d43c2be#diff-a5fbadd39b33b572d10ca8c7fbc6f06e6e85df1b725e4a8833d115d95a3a55b5R38), right?